File: day9_part1.py
import sys, os

def parse_input(input_text):
    try:
        coords = []
        for line in input_text.splitlines():
            x, y = line.strip().split(',')
            x, y = int(x), int(y)
            coords.append((x, y))
        return coords
    except ValueError:
        print(f"Error parsing line: {line}")

def find_result(coords):
    max_area = 0
    for x1, y1 in coords:
        for x2, y2 in coords:
            if x1 != x2 and y1 != y2:
                area = (abs(x2 - x1) + 1) * (abs(y2 - y1) + 1)
                if area > max_area:
                    max_area = area
    return max_area

if __name__ == "__main__":
    # If no filename is provided, try common input filenames before failing.
    if len(sys.argv) == 2:
        input_filename = sys.argv[1]
    else:
        # Prefer Gayle's input if present, otherwise Dean's input
        if os.path.exists("day9_input_gayle.txt"):
            input_filename = "day9_input_gayle.txt"
        elif os.path.exists("day9_input_dean.txt"):
            input_filename = "day9_input_dean.txt"
        else:
            print(f"Usage: python {sys.argv[0]} <input_filename>")
            sys.exit(1)
  
    input_text = open(input_filename).read()
    coords = parse_input(input_text)     
    result = find_result(coords)
    print(f"Day 9 Part 1 result: {result}")

File: day9_part1.rb
DEBUG=false

def read_file(file_path)
    coordinates = []
    File.open(file_path, "r") do |file|
        lines = File.readlines(file_path).map(&:chomp)
        coordinates = lines.map{|line| line.split(",").map{|s| s.to_i } }
    end
    return coordinates
end

def calc_area(coordinate1, coordinate2)
    width = (coordinate2[0] - coordinate1[0]).abs + 1
    height = (coordinate2[1] - coordinate1[1]).abs + 1
    return width * height
end

def find_largest_rectangle_area(coordinates)
    max_area = 0
    while !coordinates.empty? do 
        first_coordinate = coordinates.shift
        coordinates.each do |second_coordinate| 
            area = calc_area(first_coordinate, second_coordinate)
            puts "area: #{area} with coords: #{first_coordinate}, #{second_coordinate}" if DEBUG
            if area > max_area
                max_area = area
                puts "new max_area: #{max_area} with coords: #{first_coordinate}, #{second_coordinate}" if DEBUG
            end
        end
    end
    return max_area
end

if __FILE__ == $0
    filename = ARGV[0] || 'day9_input_gayle.txt'
    coordinates = read_file(filename)
    largest_rectangle = find_largest_rectangle_area(coordinates)
    puts "largest_rectangle: #{largest_rectangle}"
end